### PR TITLE
Blockchain basics part 2

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/introduction-to-smart-contracts.rst [hunk 1]
```diff
@@ -397,13 +400,17 @@ Instruction Set
...
-All instructions operate on the basic data type, 256-bit words.
+incorrect or inconsistent implementations which could cause consensus problems.
+All instructions operate on the basic data type, 256-bit words or on slices of memory
+(or other byte arrays).
 The usual arithmetic, bit, logical and comparison operations are present.
 Conditional and unconditional jumps are possible. Furthermore,
 contracts can access relevant properties of the current block
 like its number and timestamp.
 
+For a complete list, please see the list of opcodes_ as part of the inline
```
Comment (leonardoalt) [inline docs/introduction-to-smart-contracts.rst]: Docs tests are failing on `opcodes`

File: docs/introduction-to-smart-contracts.rst [hunk 2]
```diff
@@ -457,21 +467,21 @@ Logs
 
 It is possible to store data in a specially indexed data structure
 that maps all the way up to the block level. This feature called **logs**
-is used by Solidity in order to implement **events**.
+is used by Solidity in order to implement events_.
```
Comment (leonardoalt) [inline docs/introduction-to-smart-contracts.rst]: Should `events` point somewhere?

**Comment by chriseth** [comment]
Hopefully it works now.

**Comment by chriseth** [comment]
Ok, omitting ref only seems to work if the target is in the same document.

File: docs/introduction-to-smart-contracts.rst [hunk 3]
```diff
@@ -387,7 +389,8 @@ the topmost 16 elements to the top of the stack or swap the
 topmost element with one of the 16 elements below it.
 All other operations take the topmost two (or one, or more, depending on
```
Comment (ekpyron) [inline docs/introduction-to-smart-contracts.rst:404]: ``the topmost two (or one, or more`` :-)? Maybe ``All other operations take the elements they require from the top of the stack and push the result onto the stack``.

**Review by ekpyron — DISMISSED** [review]
Small suggestion; otherwise looks fine to me.

File: docs/introduction-to-smart-contracts.rst (hunk 3 above)
Comment (chriseth) [reply docs/introduction-to-smart-contracts.rst:404]: Feel free to create a PR :)

**Comment by chriseth** [comment]
Rebased.

**Pull request merged**